### ISSUE TITLE
Internationalisation

### ISSUE THREAD

**Issue by wooorm** [get-alex/alex#202]
### Subject of the issue

Alex, for now, supports just English, but retext could work with other Latin-script languages.

* We could support other languages in [`retext-profanities`](https://github.com/retextjs/retext-profanities), as there are several open lists of potential profanities in other languages
* We could support other languages in [`retext-equality`](https://github.com/retextjs/retext-equality), but that would require people to hand-translate the phrases

### Do you know a language other than English

...and are you able (it’ll take some time) and wiling to help? Create a new issue for your language and let’s start working on this!

**Comment by CRomano31415** [comment]
I see the issue in [retext-profanities](https://github.com/retextjs/retext-profanities/issues/8). I'll start putting a list in Spanish together 👍🏼

**Comment by JekRock** [comment]
Does it support only Latin languages? I can help with Russian and Ukrainian.

**Comment by deatheguard** [comment]
I can help with French and/or English!

**Comment by wooorm** [comment]
@CRomano31415 Please do! Feel free to open a new issue / PR about Spanish

@JekRock Under the hood, mainly Latin-script languages are supported. Cyrillic could potentially work. See [`parse-latin`](https://github.com/wooorm/parse-latin) for the main parts about it.

@deatheguard French would be great. Looks like you found GH-207 already. For English, you can just help with [`retext-profanities`](https://github.com/retextjs/retext-profanities) and [`retext-equality`](https://github.com/retextjs/retext-equality) already!

@baezor Please do as well! Let’s check if @CRomano31415 opens an issue so y’all can collaborate :)

**Comment by AhmedRedaAmin** [comment]
Can I add support for Arab Latin-Script Slang ? (Used heavily on Social media)
Note : it includes numbers for some Arabic sounds , if that will cause problems for `parse-latin`

**Comment by wooorm** [comment]
@AhmedRedaAmin Yes, feel free to do that, and create a new issue for it! 

> Note : it includes numbers for some Arabic sounds , if that will cause problems for parse-latin

I’m not sure, I don’t know Arabic, let’s try it out!

**Comment by inesbenomar18** [comment]
I can help with Arabic and French as well! :)

**Comment by wooorm** [comment]
@inesbenomar18 Awesome! For french, start with GH-207. And feel free to open a new issue about Arabic (note though that I think Arabic-script may not be properly supported)

**Comment by AhmedRedaAmin** [comment]
Okay , so I noticed there were 2 approaches to adding new languages , so I 'll take @CRomano31415 's approach , I already started by borrowing her repo template , I hope you don't mind Claudia !

**Comment by wooorm** [comment]
@rampagesang Awesome! Unfortunately I don’t think languages that use a script other than the Latin-script can work with the current technical setup :(

**Comment by Yangeok** [comment]
Can i help you to translate language korean?

**Comment by wooorm** [comment]
@Yangeok Hey, that’s cool! I don’t think Korean will work (see the comment before yours), but German would definitely work!

**Comment by PaoloWeishaupt** [comment]
I can help with italian, spanish, french and german.

**Comment by luigicorreia** [comment]
I can help with Portuguese.

**Comment by toucedam** [comment]
> I can help with spanish!. Is someone working on spanish here?

yes

**Comment by GledsonAfonso** [comment]
I can help with Portuguese too (pt-BR).

**Comment by waaghree** [comment]
> ### Do you know a language other than English
> ...and are you able (it’ll take some time) and wiling to help? Create a new issue for your language and let’s start working on this!

How about Urdu?  It's in mostly Arabic script with added characters, however, for a large part people use English characterset to write on social media in something called "Roman Urdu".

**Comment by wooorm** [comment]
@waaghree Yep, romanised can work. For inspiration, see the Arabic cuss file added in https://github.com/words/cuss/pull/16!

**Comment by AhmedRedaAmin** [comment]
@waaghree yes , I worked on something very similar . You can refer to @wooorm 's link as well as the issue titled Arabic Latin-Script on this repo , feel free to ping me if you want to ask about anything .
Glad to see Arabic script being transliterated for more than one language :D , Good luck mate.

**Comment by GledsonAfonso** [comment]
@wooorm @AhmedRedaAmin  Except from the words/cuss project, there's another way that we can help with Internationalisation? If yes, how?

**Comment by AhmedRedaAmin** [comment]
@GledsonAfonso  So here is how it works , you add the words to the words/cuss project , rate them based on profanity , then they get used in get-alex/alex to detect profane words , that is as far as I know how the Yaml files can Identify the inappropriate words , as for the "insensitive" phrases , they get added to retext-inequality by hand directly together with the suggested alternatives , they are different from flat out cusses which obviously don't have alternatives suggested .
This is what I understood atleast .
**Short answer** : The second best way to help  with internationalization is to add your native language insensitive phrases to `retext-inequality` along with their suggested alternatives , Good luck mate ! :D

**Comment by GledsonAfonso** [comment]
@AhmedRedaAmin  Thank you for the explanation! I think it would be interesting if we had some section in the get-alex/alex project (wiki maybe? With a mention in the README file and all the cool stuff) about this... or this already exists and I'm just making a fool of myself here, haha.

Anyway, thanks again for the heads up. I'll see if I can add some phrases in retext-equality as soon as I get the gist of it. Cheers!

**Comment by wooorm** [comment]
@GledsonAfonso We could definitely use a section on internationalisation in the [readme](https://github.com/get-alex/alex/blob/master/contributing.md#contributions)!

**Comment by GledsonAfonso** [comment]
@wooorm Great! Do we need to create an issue for that?

**Comment by wooorm** [comment]
Sure, you could create a separate issue, or if you’d like to work on it feel free to take a stab at it!

**Comment by OtacilioN** [comment]
Hey Folks, I would love to help with Portuguese!

**Comment by GledsonAfonso** [comment]
@wooorm Okay!

@OtacilioN Thanks for creating the issue!
